execution/execution.go

```diff
@@ -45,40 +45,49 @@ const stepsBatch = 10
 
 // New creates new physical query execution for a given query expression which represents logical plan.
 // TODO(bwplotka): Add definition (could be parameters for each execution operator) we can optimize - it would represent physical plan.
-func New(expr parser.Expr, storage storage.Queryable, mint, maxt time.Time, step, lookbackDelta time.Duration) (model.VectorOperator, error) {
+func New(expr parser.Expr, queryable storage.Queryable, mint, maxt time.Time, step, lookbackDelta time.Duration) (model.VectorOperator, error) {
 	opts := &query.Options{
 		Start:         mint,
 		End:           maxt,
 		Step:          step,
 		LookbackDelta: lookbackDelta,
 		StepsBatch:    stepsBatch,
 	}
-	selectorPool := engstore.NewSelectorPool(storage)
-	return newCancellableOperator(expr, selectorPool, opts)
+	selectorPool := engstore.NewSelectorPool(queryable)
+	hints := &storage.SelectHints{
+		Start: mint.UnixMilli(),
+		End:   maxt.UnixMilli(),
+		Step:  step.Milliseconds(),
```

Comment:
Maybe we can add a TODO here about this: https://github.com/prometheus/prometheus/pull/11163/files#diff-ad2761bb162a2f12326b18f2714ef54f4cd476cfe3388f77b2b20c5ff82e4faaR857? Subqueries are not supported at the moment so we can't add this but maybe we can add a comment so that we would not forget.

Reply:
Makes sense, I added a todo for this. I also copied the tests from the old engine from the latest main, so that they fail if we don't adjust the step hint properly.